dashboard/app/models/concerns/curriculum/course_types.rb

```diff
@@ -1,8 +1,14 @@
 # Model concern for common curriculum audience methods
 # To use, include in a model and call the desired method.
-module Curriculum::CourseAudiences
+module Curriculum::CourseTypes
```

Comment:
please help me understand the thinking behind this name -- did you have an idea in mind for whether this concern would (a) be used as a catch-all for all cases where `Script` and `UnitGroup` share functionality, or whether you are (b) grouping together only a specific subset of functionality (more or less what you are working on now)?

my mental model is that course types is a slightly different way of slicing, dicing and grouping our courses, e.g. CSF, CSD, CSP, CSC, HOC, Self-paced PL, live PL, etc each being their own type, whereas audience (and self-paced vs teacher-led) are a different set of dimensions to slice along. maybe you are seeing more overlap between these two ways of categorizing than I am?

Reply:
The name is has to do with the proposal Tess is currently sharing with the other PMs where the intersection of participant audience and instruction type tells you the type of a course. So of your options this is (b).

To me CSF, CSD, CSP are course offerings. Where as self-paced pl, virtual pl, etc are course types. Course offerings fit into different course types. Take CSP and CSA for example. We are going to want a lot of the same behavior for both of these courses because they are  *Teacher-Led* courses for *High School Students*.  CSC is especially interesting as we look at how to handle it going forward. Right now we check curriculum umbrella to make decisions like what type of instructions to show or when to show example solutions. However with CSC we are planning to have high school CSC modules as well as elementary school CSC modules so checking curriculum umbrella won't work. However if a CSC module is marked with instruction type and participant audience we will be able to determine how to show it in the future (This relies on the idea that we would split student audience into probably 3 audiences: high school, middle school, elementary school). 

I drew out my thought on this here: https://docs.google.com/spreadsheets/d/1SOFfIsAUkKsbG2ZvTUBaev9dVk-aisE7aTL6i1Mx9ZY/edit#gid=0

Reply:
very helpful, thank you Dani!

Reply:
I like where we're heading with separating _participant audience_ and _instruction type_.  One thing that I think we should be mindful about is to have logic in just one dimension wherever possible to reduce the matrix of cases that behave differently.  If we find that we often need to have different behavior based on two (or more) dimensions, then I think we've still organized our courses better but we haven't really reduced the complexity of the system (which I think we really need to do for long-term health).

Reply:
cc @tess323

Reply:
thanks for the cc @dmcavoy - super interesting lets discuss this when we sync next about this categorization.  Copying James' note into our 1:1 doc